modules/bibdocfile/lib/bibdocfile.py

```diff
@@ -3851,43 +3854,52 @@ def download_external_url(url, docformat=None):
     from_file, to_file, tmppath = None, None, ''
 
     try:
+        while r.get('hostname_cache.%s' % hostname):
+            # We have accessed this host within the last 2 seconds
+            time.sleep(2)
         from_file = open_url(url)
     except InvenioBibdocfileUnauthorizedURL, e:
         raise StandardError, str(e)
     except urllib2.URLError, e:
         raise StandardError, 'URL could not be opened: %s' % str(e)
 
-    if not docformat:
-        # We could not determine the format from the URL, so let's try
-        # to read it from the HTTP headers.
-        docformat = get_format_from_http_response(from_file)
-
+    # We use redis as a semaphore.
+    r.set('hostname_cache.%s' % hostname, True)
```

Comment:
I was thinking yes, because in this way we avoid parallel downloads as well. Basically I am blocking downloading from the given hosts for the entire duration of the download + 2 seconds.